azkaban-web-server/src/main/java/azkaban/imagemgmt/servlets/ImageRampupServlet.java

```diff
@@ -0,0 +1,226 @@
+/*
+ * Copyright 2020 LinkedIn Corp.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package azkaban.imagemgmt.servlets;
+
+import static azkaban.Constants.ImageMgmtConstants.IMAGE_TYPE;
+
+import azkaban.imagemgmt.dto.ImageMetadataRequest;
+import azkaban.imagemgmt.exeception.ImageMgmtException;
+import azkaban.imagemgmt.exeception.ImageMgmtInvalidPermissionException;
+import azkaban.imagemgmt.exeception.ImageMgmtValidationException;
+import azkaban.imagemgmt.models.ImageRampupPlan;
+import azkaban.imagemgmt.services.ImageRampupService;
+import azkaban.server.HttpRequestUtils;
+import azkaban.server.session.Session;
+import azkaban.user.Permission.Type;
+import azkaban.utils.JSONUtils;
+import azkaban.webapp.AzkabanWebServer;
+import azkaban.webapp.servlet.LoginAbstractAzkabanServlet;
+import com.linkedin.jersey.api.uri.UriTemplate;
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.Map;
+import java.util.Optional;
+import javax.servlet.ServletConfig;
+import javax.servlet.ServletException;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+import org.apache.http.HttpStatus;
+import org.codehaus.jackson.map.ObjectMapper;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This servlet exposes the REST APIs such as create, get and update rampup for image type. Below
+ * are the supported APIs. Create Image Rampup API: POST /imageRampup?session.id=? --data
+ *
+ * @payload.json Get active rampup for an image type : GET /imageRampup/spark_job?session.id=?
+ * Update Image Rampup API: POST /imageRampup/spark_job?session.id=? --data @payload.json
+ */
+public class ImageRampupServlet extends LoginAbstractAzkabanServlet {
+
+  private static final String BASE_IMAGE_RAMPUP_URI = "/imageRampup";
+  private static final String IMAGE_RAMPUP_PLAN_ID_KEY = "id";
+  private static final UriTemplate CREATE_IMAGE_RAMPUP_URI_TEMPLATE = new UriTemplate(
+      String.format("/imageRampup/{%s}", IMAGE_RAMPUP_PLAN_ID_KEY));
+  private static final UriTemplate UPDATE_IMAGE_RAMPUP_URI_TEMPLATE = new UriTemplate(
+      String.format("/imageRampup/{%s}", IMAGE_TYPE));
```

Comment:
This URI is not correct, we shouldn't NOT use a deployable name in the URI we should use the ramp up plan ID. If the reason for this is IDs not been easy to remember then we can create alias URIs for mitigating the issue but that would be an additional URI, not a replacement for the canonical URI.

Reply:
Rampup is done for an image type and hence, the uri is based on image type. There will be one active rampup plan for an image type and all the checks are done based on that. Also it is easier for user to provide image type. Moreover, all the access and permissions are based on image type.